### DISCUSSION TITLE
Get 'Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.1.2".' after upgrade to newest version

### DISCUSSION THREAD

**Discussion by blwtxc** [kimai/kimai#4112] [On-Premise: Installation & Updates] [answered]
### Describe the issue

Followed the instructions to update/upgrade my kimai instance here

https://www.kimai.org/documentation/updates.html

Composer first complained about 'Call to undefined function', but after updating composer and running it with php8.2, that issue disappeared and the update ran succesfully.

* I run (and ran ... so this is not a major update) PHP-FPM 8.2
* I changed the default PHP version to 8.2 with `update-alternatives`

But when I open the application in a browser, everything I see is a blank page with the following error message:

Composer detected issues in your platform: Your Composer dependencies require a PHP version ">= 8.1.2".

I already:

* retried composer on command line (no issues there)
* cleared caches of
  * composer
  * kimai (with `bin/console`)
* restarted the fpm process
* updated the php version to 8.2.1 in `composer.json`
* added the flag `"platform-check": false` in `composer.json`
* re-set owner/permissions on the application dir for good measure

without any luck...

What could be the problem here?

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kimai/kimai/discussions).

### Kimai version

2.0.26

### How do you run Kimai?

Virtual Server or alike

### Which PHP version are you using?

8.2

### Logfile

`prod.log`

Only message there is the following, because I tried to execute `cache:pool:clear` -- in my attempts to fix the issue -- without providing a 'pool'

console.CRITICAL: Error thrown while running command "cache:pool:clear". Message: "Not enough arguments (missing: "pools")." {"exception":"[object] (Symfony\\Component\\Console\\Exception\\RuntimeException(code: 0): Not enough arguments (missing: \"pools\"). at /opt/kimai/kimai2/vendor/symfony/console/Input/Input.php:70)","command":"cache:pool:clear","message":"Not enough arguments (missing: \"pools\")."} []

**Comment by kevinpapst** [comment, marked as answer]
1. Do not try to trick composer, revert that
2. You are NOT running the correct PHP Version in your webserver
3. CLI and WEB are two different PHP versions, which can be configured independently.

**Reply by blwtxc** [reply]
Man, you're good!
It really was a mismatch in the version I used in my FPM process.
Thank you so much for your quick response.
Cheers
